logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationUseCase.kt

```diff
@@ -187,21 +189,32 @@ internal class JoinExistingMLSConversationUseCaseImpl(
 
             type == Conversation.Type.Self -> {
                 kaliumLogger.d("$TAG: Establish Self MLS Conversation ${conversation.id.toLogString()}")
-                mlsConversationRepository.establishMLSGroup(
-                    protocol.groupId,
-                    emptyList()
-                ).onSuccess {
-                    kaliumLogger.logStructuredJson(
-                        level = KaliumLogLevel.INFO,
-                        leadingMessage = "Establish Group",
-                        jsonStringKeyValues = mapOf(
-                            "conversationId" to conversation.id.toLogString(),
-                            "conversationType" to type,
-                            "protocol" to ConversationOptions.Protocol.MLS.name,
-                            "protocolInfo" to conversation.protocol.toLogMap(),
+                clientRepository.getClientsByConversationId(conversation.id)
+                    .flatMap { clients ->
+                        if ((clients[selfUserId]?.size ?: 0) > 1) {
```

Comment:
```suggestion
                    .flatMap { clients ->
                        val hasOtherClients = (clients[selfUserId]?.size ?: 0) > 1
                        if (hasOtherClients) {
```

Reply:
After discussing with @typfel we can get rid of this 😄